### PR TITLE
Refactored asset and token management

### PR DESCRIPTION
**Pull request by theblockstalk**
1. created centralized registry of supported tokens in `./src/util/tokenRegistry.ts`
    - moved LEOS asset to the registry and removed bespoke logic
    - centralized asset creation and balance updates
    - other helper functions for chains and tokens
2. Fixed Antelope block explorer logic and key management
3. Added `isTestnet()` and `isTransferable()`  and comparison/sum members to chain and token for better abstraction
4. New `PangeaVestedToken extends AntelopeToken` which adds extra vesting logic
5. added branch protection in VS code and fixed README.md issues
6. Small fixed to Privacy Policy

TODO:
- allow send Antelope token src/containers/SendAssetContainer.tsx

### CODE REVIEW

File: src/utils/chain/types.ts [hunk 1]
```diff
@@ -244,13 +249,15 @@ export abstract class AbstractToken implements IToken {
...
+    protected transferable = true;
 
-    constructor(name: string, symbol: string, precision: number, chain: IChain, logoUrl: string) {
+    constructor(name: string, symbol: string, precision: number, chain: IChain, logoUrl: string, transferable = true) {
         this.name = name;
         this.symbol = symbol;
         this.precision = precision;
         this.chain = chain;
         this.logoUrl = logoUrl;
+        this.transferable = transferable;
```
Comment (sadiabbasi) [inline src/utils/chain/types.ts:260]: what the purpose/means of transferable?

File: src/utils/tokenRegistry.ts [hunk 2]
```diff
@@ -0,0 +1,216 @@
...
+    } catch (error) {
+        if (error.message.startsWith('Key not found for')) return null;
+        throw error;
+    }
+}
+
+export async function getAccountFromChain(chainEntry: TokenRegistryEntry, user: IUser): Promise<IAccount> {
+    const { chain, token } = chainEntry;
+    const asset = await assetStorage.findAssetByName(chainEntry.token);
+    const key = await getKeyFromChain(chainEntry);
```
Comment (sadiabbasi) [inline src/utils/tokenRegistry.ts:151]: do we need to add condition here like if(!key) or if(!asset)

File: src/utils/tokenRegistry.ts [hunk 3]
```diff
@@ -0,0 +1,216 @@
...
+export async function getKeyOrNullFromChain(chainEntry: TokenRegistryEntry): Promise<IPrivateKey | null> {
+    try {
+        return await getKeyFromChain(chainEntry);
+    } catch (error) {
+        if (error.message.startsWith('Key not found for')) return null;
+        throw error;
+    }
+}
+
+export async function getAccountFromChain(chainEntry: TokenRegistryEntry, user: IUser): Promise<IAccount> {
```
Comment (sadiabbasi) [inline src/utils/tokenRegistry.ts:148]: we can pass user as an optional params? as we don't need this if chainType is ethereum

File: src/containers/AssetDetailContainer.tsx [hunk 4]
```diff
@@ -80,57 +54,66 @@ const AssetDetailContainer = (props: AssetDetailProps) => {
         <View style={styles.container}>
             <View style={styles.content}>
                 <ScrollView contentContainerStyle={styles.scrollViewContent}>
-                    <View style={styles.header}>
-                        <View style={styles.networkHeading}>
-                            <Image source={asset.icon || Images.GetImage('logo1024')} style={styles.faviconIcon} />
-                            <View style={styles.assetsNetwork}>
-                                <Text style={styles.assetsNetworkText}>{asset.network}</Text>
+                    {asset && (
```
[the remark this answers is not in the document]
Comment (theblockstalk) [reply src/containers/AssetDetailContainer.tsx:57]: should not be necessary. We fetch the asset object async from local storage (no network fetches) so should be near instant. This check here is just to have type safety.

File: src/utils/chain/types.ts (hunk 1 above)
Comment (theblockstalk) [reply src/utils/chain/types.ts:260]: this is used by `isTransferable()` which tells the AssetDetails screen if it should display a message `"All {asset.token.symbol} is vested until the public sale"`

File: src/utils/tokenRegistry.ts (hunk 2 above)
Comment (theblockstalk) [reply src/utils/tokenRegistry.ts:151]:
```ts
const asset = await assetStorage.findAssetByName(chainEntry.token);
// condition is checked a few lines later
const key = await getKeyFromChain(chainEntry);
// always returns a private key or throws an error
```

File: src/utils/tokenRegistry.ts (hunk 3 above)
Comment (theblockstalk) [reply src/utils/tokenRegistry.ts]: fixed

File: src/containers/ReceiveAssetContainer.tsx [hunk 5]
```diff
@@ -8,33 +8,31 @@ import { useEffect, useState } from 'react';
...
+    chain: IChain;
 };
 
 const ReceiveAssetContainer = (props: ReceiveAssetProps) => {
     const [showPopover, setShowPopover] = useState(false);
-
-    const [asset, setAsset] = useState<AccountDetails | null>(null);
+    const [asset, setAsset] = useState<AccountTokenDetails | null>(null);
     const [loading, setLoading] = useState<boolean>(true);
+    const networkLogo = asset?.token.icon ? { uri: asset.token.icon } : Images.GetImage('logo1024');
```
[the remark this answers is not in the document]
Comment (theblockstalk) [reply src/containers/ReceiveAssetContainer.tsx]: fixed

File: src/containers/LoginPassphraseContainer.tsx [hunk 6]
```diff
@@ -8,16 +8,37 @@ import TInfoBox from '../components/TInfoBox';
...
+import { Asset } from '../utils/chain/types';
+import { assetStorage } from '../utils/StorageManager/setup';
 import TSpinner from '../components/atoms/TSpinner';
+import Debug from 'debug';
+
+const debug = Debug('tonomy-id:containers:LoginPassphraseContainer');
 
 const tonomyContract = TonomyContract.Instance;
 
+export async function addNativeTokenToAssetStorage(user: IUserBase) {
```
[the remark this answers is not in the document]
Comment (theblockstalk) [reply src/containers/LoginPassphraseContainer.tsx]: fixed

File: src/containers/AssetDetailContainer.tsx [hunk 7]
```diff
@@ -4,42 +4,28 @@ import { Images } from '../assets';
...
-
-            setAccountName(accountName);
-        };
-
-        if (user) {
-            fetchAccountName();
-        }
-    }, [user]);
+    const networkLogo = asset?.token.icon ? { uri: asset.token.icon } : Images.GetImage('logo1024');
+    const { account } = useChainWalletStore();
```
[the remark this answers is not in the document]
Comment (theblockstalk) [reply src/containers/AssetDetailContainer.tsx]: fixed

**Pull request merged**